pkg/framework/proxies.go

```diff
@@ -1,330 +1,389 @@
+)
+
+const (
+	mapiControllersDeploymentName         = "machine-api-controllers"
+	machineControllerContainerName string = "machine-controller"
+	proxyNamespace                        = MachineAPINamespace
+	proxyName                             = "mitm-proxy"
+	mitmSignerName                        = "mitm-signer"
+	mitmBootstrapName                     = "mitm-bootstrap"
+	mitmCustomPKIName                     = "mitm-custom-pki"
+	mitmCustomPKINamespace                = "openshift-config"
+	mitmDaemonsetName                     = proxyName
+	mitmServiceName                       = proxyName
 )
 
 const proxySetup = `
 cd /.mitmproxy
 cat /root/certs/tls.key /root/certs/tls.crt > /.mitmproxy/mitmproxy-ca.pem
 curl -O https://snapshots.mitmproxy.org/5.3.0/mitmproxy-5.3.0-linux.tar.gz
 tar xvf mitmproxy-5.3.0-linux.tar.gz
-HOME=/.mitmproxy ./mitmdump
+HOME=/.mitmproxy ./mitmdump --ssl-insecure
 `
 
-const mitmSignerCert = `
------BEGIN CERTIFICATE-----
-MIICQDCCAamgAwIBAgIUXyjCAhFKon7pffzuFrHRBmLQRqwwDQYJKoZIhvcNAQEL
-BQAwMjELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAk5DMRYwFAYDVQQDDA1taXRtLXBy
-b3h5LWNhMB4XDTIwMTEyMDIxMTk1NVoXDTQ4MDUwMTIxMTk1NVowMjELMAkGA1UE
-BhMCVVMxCzAJBgNVBAgMAk5DMRYwFAYDVQQDDA1taXRtLXByb3h5LWNhMIGfMA0G
-CSqGSIb3DQEBAQUAA4GNADCBiQKBgQDPqHuUZz1Qt236e03XWjsVCy1OnuuWQ3a1
-OQX21waqypGPem6iKiSmAmf+YrKdkX2O5L2sqPjpZ+civ9z5h9d0xCCNy+06UQTu
-+pIrgIxm2p5wQmSGgI5KNUL6dgU3dC5aRDMU7hV0RGXKrJktTmU2SnZsOUe4SZ5L
-Vyanpk8N1QIDAQABo1MwUTAdBgNVHQ4EFgQUFD2D/LygThvwLRKukmH1Y++hxLIw
-HwYDVR0jBBgwFoAUFD2D/LygThvwLRKukmH1Y++hxLIwDwYDVR0TAQH/BAUwAwEB
-/zANBgkqhkiG9w0BAQsFAAOBgQCLZeGgfFRCOWqSXcDzUzeJE03HD/y2rLY+hDMn
-kj69xY3+K0z4DRbb1vpMiYsTYF/Z1d6blND4KW+7oi9R6PPYg6XATDMf9tVfTQIe
-qBkDNfqABPlwJABpixwD20XXQUBqADEyO3tdQLtiMi5Qr6QHOX3+FepiHAgdxAFt
-Mqy4Gw==
------END CERTIFICATE-----
-`
-const mitmSignerKey = `
------BEGIN RSA PRIVATE KEY-----
-MIICXgIBAAKBgQDPqHuUZz1Qt236e03XWjsVCy1OnuuWQ3a1OQX21waqypGPem6i
-KiSmAmf+YrKdkX2O5L2sqPjpZ+civ9z5h9d0xCCNy+06UQTu+pIrgIxm2p5wQmSG
-gI5KNUL6dgU3dC5aRDMU7hV0RGXKrJktTmU2SnZsOUe4SZ5LVyanpk8N1QIDAQAB
-AoGBAJksuGuRc8MUawV26sZNgoNVSUhTJYgjn26x71pS5lIZNiHVt8HawEnMQJV+
-jC56YVmEFP1FbsYMpIwXZpKRxzS/uXInKUbRlYQuJ8kVG7a0foe+vja1gxmG4b+Z
-69V6S+4Y8AhHEcw+Ek04LUWOzEV0NpBa8VJYVHtW3bTarE0JAkEA9DBvOmkg9rit
-x2M9zLR+BCPvgP16EARY2Ik+ZlOP7hC2iPL9mEjxSujJ78SSrrY09hPomemAqBz6
-eiWbsL9KgwJBANmzt+pOGu0fON3MFbjRKfZHAdsDJQipTEGKr5jig8nZg84R+zBA
-rfm9Vm2zdHaZ6rGTpMv62roHrXnq9y1htscCQQDQo8GlqsWbiNgSkNzw1xcE+p9d
-Gzb8EHrJKRrD24oS4vzTrqq3PzvLwXMpBlA+Lzi5OPF48GYZPglV7GRGdGt5AkAY
-7v53dW6cDeFjdcZfHoWh0UwjG18YeNtk/k9SQU86xRDVfzW3txC187t8YPtLwiEh
-KXnMavS2Lb7uobyhk/ltAkEA48NjOS7IL1JMR2X/r5uBrEQhd/XjnaYePdav32Ii
-2ygHNMMkN1/ueVBCypFaU2UZDVNabyZ+MJYnR8dK4xxpQg==
------END RSA PRIVATE KEY-----
-`
+// DeployProxy deploys a MITM Proxy to the cluster.
+func DeployProxy(c client.Client, gomegaArgs ...interface{}) {
+	ctx := context.Background()
+	komega.SetClient(c)
+	komega.SetContext(ctx)
```

Comment:
Rather than setting the default komega, should we create our own komega instance to use within this function?

Modifying the global state could lead to weird issues in other tests later down the line I would be concerned